lib/Github/Api/PullRequest.php

```diff
@@ -163,6 +164,10 @@ public function merge($username, $repository, $id, $message, $sha, $mergeMethod
             $mergeMethod = $mergeMethod ? 'squash' : 'merge';
         }
 
+        if (!in_array($mergeMethod, array('merge', 'squash', 'rebase'), true)) {
+            throw new InvalidArgumentException(sprintf('"$mergeMethod" must be one of ["merge", "squash", "rebase"] ("%s" given).', $mergeMethod));
```

Comment:
It is better to fail early. Since the API client **knows** that the request is going to fail, why now throw an exception right away?